File: julia/analytical_functions.jl
function inf_sum(x, t)
    tol = 1e-10
    k = 1
    max_k = 100
    eps = 1.0
    f = 0.0
    f_old = 1.0
    T = 0.0
    
    while eps > tol && k < max_k
        A = 2k - 1
        f = (4 / (A * π)) * sin.(A * π * x) * exp(-(A)^2 * π^2 * t)
        T = T .+ f
        eps = maximum(abs.(T .- f_old))
        k += 1
        f_old = T
    end
    
    return T
end

File: python/analytical_functions.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Created on Wed Nov 29 13:41:52 2023

@author: carlosar
"""

import numpy as np

def inf_sum(x,t):
    '''
    Infinite sum in analytical solution approximated with max k sum
    terms or if the term is below some tolerance 
    '''
    tol = 10.**(-10.)
    k = 1
    max_k = 100
    eps = 1.
    f = 0.
    f_old = 1.0
    T = 0.
    while (eps > tol) and (k<max_k):
        A = ((2*k)-1)
        f = (4/(A*np.pi))*(np.sin(A*np.pi*x))*np.exp(-(A)**2*np.pi**2*t)        
        T = T+f
        eps = np.max(np.abs(T-f_old))
        k = k+1
        f_old = T
    return T